dockerfiles/stages/1-build-deps

```diff
@@ -0,0 +1,121 @@
+#################################################################################################
+# The "build-deps" stage
+# - Installs all compilers/interpreters, tools, and OS packages on debian ${deb_codename}-slim
+#################################################################################################
+ARG deb_codename=stretch
+FROM debian:${deb_codename}-slim AS build-deps
+
+# Ocaml Version
+ARG OCAML_VERSION=4.11
+ARG OCAML_REVISION=.2
+ARG OCAML_VARIANT=
+ARG OCAML_PACKAGE=
+ARG OPAM_VERSION=2.0.7
+
+# Golang version number used to detemine tarball name
+ARG GO_VERSION=1.16.5
+
+# Rust Version passed into rustup-init, can also be "stable", "nightly" or similar
+ARG RUST_VERSION=1.52.1
+
+# Rocksdb commit tag/branch to clone
+ARG ROCKSDB_VERSION=v5.17.2
+
+# OS package dependencies
+# First add support for https and pkg-config for apt, then install everything else
+# TODO: make sure this is the minimum compile-time deps
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update \
```

Comment:
Shouldn't we remove the apt cache after this step?

Reply:
Same with other stages that do apt-get installs.

Reply:
No actually, in `slim` debian images apt is already configured to skip creating/maintaining that cache in the first place so we don't need to delete it afterwards.